Riot/Modules/Onboarding/AuthenticationCoordinator.swift

```diff
@@ -0,0 +1,418 @@
+// File created from ScreenTemplate
+// $ createScreen.sh Onboarding Authentication
+/*
+ Copyright 2021 New Vector Ltd
+ 
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+ 
+ http://www.apache.org/licenses/LICENSE-2.0
+ 
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+import UIKit
+import MatrixSDK
+
+@available(iOS 14.0, *)
+struct AuthenticationCoordinatorParameters {
+    let navigationRouter: NavigationRouterType
+    /// The screen that should be shown when starting the flow.
+    let initialScreen: AuthenticationCoordinator.EntryPoint
+    /// Whether or not the coordinator should show the loading spinner, key verification etc.
+    let canPresentAdditionalScreens: Bool
+}
+
+/// A coordinator that handles authentication, verification and setting a PIN.
+@available(iOS 14.0, *)
+final class AuthenticationCoordinator: NSObject, AuthenticationCoordinatorProtocol {
```

Comment:
This file could use splitting up, GitHub did not even want to show it to me at first. I think the coordinator is doing too much low level work, and the important bits are probably hard to test, see for example the comment below about state setup.

Reply:
I've split out the key verification listener as mentioned in the other comment. I'm planning to remove the extension at the end of the file when handling either SSO or the removal of the legacy flow. Happy to simplify further if you can see any further options (to my eyes, the remaining functionality all seems related to navigation through the flow so not sure how to further reduce the size).